Rework installer to replace the use of pungi(legacy)

fepitre opened the pull request.

fepitre committed:
Makefile: drop pungi-legacy

Replace pungi(legacy) by creating specific targets:
1) pungi-gather for collecting required RPM from comps on the targeted ISO
2) lorax for creating runtime installer (anaconda)
3) mkisofs + isohybrid + implantisomd5

This is to handle deprecated pungi-legacy on upstream and to not use pungi-koji
which is currently overkill for our purpose.

fepitre committed:
lorax: use repo file for sources

fepitre committed:
volid: fix according to ISO

fepitre committed:
gather: greedy=none to prevent downloading useless rpm

fepitre committed:
gather: fix missing qubes groups when creating iso repo

fepitre committed:
qubes-kickstart: use metalinks

fepitre committed:
Makefile: product is QubesOS not Qubes

fepitre committed:
conf: replace mirrorlist by metalink in ks

marmarek committed:
Fix permissions on installation image

Make files readable also by non-root user.

marmarek committed:
conf: switch build configs to use dom0 fc31 repositories

QubesOS/qubes-issues#5529

fepitre committed:
pungi: drop --selfhosting option

marmarek committed:
gitignore buildinfo files too

marmarek reviewed (changes requested):
Missing items:
- [x] make it possible for lorax to use online repos too (right now only `pungi-gather` follows `$(INSTALLER_KICKSTART)`)
 - [ ] _verify_ if both pungi-gather and lorax refuse to use unsigned/invalid signed packages (see travis for test repos urls) - retrying build should still refuse such packages  - i.e. verify if those do not stay in cache bypassing verification on the next run 
- [ ] update travis config

fepitre commented:
For point one, you mean `lorax` should use repositories defined in $(INSTALLER_KICKSTART)?

marmarek commented:
> For point one, you mean `lorax` should use repositories defined in $(INSTALLER_KICKSTART)?

Something like this. Generally it should be possible to make it possible to build the ISO using online repositories. It could be done by using repositories extracted from `$(INSTALLER_KICKSTART)`, or another/additional repo file(s), or something else.

fepitre committed:
lorax: user repo file from kickstart

fepitre committed:
Move scripts under 'scripts' directory.

TODO: remove path dependency of rpm_verify in builder-rpm and infrastructure

fepitre committed:
travis: switch to dom0 Fedora 31

marmarek committed:
Install python3-jinja2 needed by scripts/ksparser

marmarek committed:
Prefer createrepo_c over createrepo

marmarek committed:
Exclude debug and source packages from installation image

marmarek committed:
travis: update test for package verification

Use two separate repositories for pungi-gather testing and lorax
testing. One contains (unsigned) package downloaded by pungi-gather
only, and the other - by lorax only.

marmarek committed:
Use proper "Qubes OS" as a product name
